app/tests/cases_tests/test_api.py

```diff
@@ -321,3 +325,38 @@ def request_processing():
     # Jobs should only be run once
     response = request_processing()
     assert response.status_code == 400
+
+
+@pytest.mark.django_db
+def test_filter_images_api_view(client):
+    alg = AlgorithmFactory()
+    user = UserFactory()
+    alg.add_editor(user=user)
+
+    alg_result = AlgorithmResultFactory(
+        job__algorithm_image__algorithm=alg, job__creator=user
+    )
+
+    im1, im2 = ImageFactory(), ImageFactory()
+    alg_result.images.add(im1)
+    alg_result.images.add(im2)
+
+    response = get_view_for_user(
+        viewname="api:image-list",
+        client=client,
+        user=user,
+        content_type="application/json",
+    )
+    assert response.status_code == 200
+    assert response.json()["count"] == 3  # [alg_result.job.image, im1, im2]
```

Comment:
It's important to check for content in code, rather than just leaving it in a comment. Why not `assert {r["pk"] for r in response.json()["results"]} == {i.pk for i in [alg_result.job.image, im1, im2]}`?